lib/housekeeping/timestamps.rb

```diff
@@ -9,16 +9,22 @@ module Housekeeping::Timestamps
     # related_class = self.name
     # related_table_name = self.table_name
 
-    scope :created_before_date, -> (date) { where('created_at < ?', "#{date}") }
+    scope :created_before_date, ->(date) { where('created_at < ?', "#{date}") }
+    scope :created_in_date_range, ->(start, c_end) { where('created_at >= ? and created_at <= ?', start, c_end) }
+    scope :updated_in_date_range, ->(start, u_end) { where('updated_at >= ? and updated_at <= ?', start, u_end) }
+    scope :first_created, -> { all.order(created_at: :asc).first }
+    scope :last_created, -> { all.order(created_at: :desc).first }
+    scope :first_updated, -> { all.order(updated_at: :asc).first }
+    scope :last_updated, -> { all.order(updated_at: :desc).first }
```

Comment:
Are these scopes returning Scopes or instances?!  A scope with limit(1) I expect?

Reply:
My bad. Originally, they were originally scopes, but I changed the way I was thinking about them. Do you think they should be scopes, or just methods?

Reply:
If we want them to return ActiveRecord::Relation then they can remain scopes, but you'd have to change them to something like {order(created_at: :asc).limit(1)}.  I'm not sure how useful that scope would be.

In present form they should be methods, since they return an instance.

Note that I don't think you need 'all'.